soda_machine.rb

```diff
@@ -7,12 +9,31 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
   end
 
   def sell(soda_brand)
+    s_soda = find_soda(soda_brand)
```

Comment:
Excellent job using the method you had previously created so as not to recreate work you have already done.